### PR TITLE
Small updates to RP Filters.

### LINKED ISSUE #16331: DI for RP Filters is a little confusing
In the opening section, this topic states that RP filters:

> * Have access to the following services:
>   * `IConfiguration`
>   * `IHostEnvironment`
>   * `IWebHostEnvironment`
> * Can **not** access services other than the preceding by Dependency Injection (DI). The `TypeFilterAttribute` and `ServiceFilterAttribute` can access the services registered in the DI container. For more information, see ServiceFilterAttribute and TypeFilterAttribute.

These services aren't provided to the filter via DI. They're just passed in to the filter's constructor when added in `Startup`. The only DI here is into the `Startup` constructor, but that's not related to RP filters and how DI works for those. I don't think there's anything wrong with passing e.g. `IConfiguration` this way, but it's not DI and RP filters don't have special access to these. I think this makes the explanation a bit confusing.

I think I'd remove the first bullet and update the second to just indicate that DI can be used and link as it already does.

At a higher level, I'd like to see this page merged into the existing filters page. I realise that might not be practical, but it's just a thought given how closely related the two topics are.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f0646a3d-a5d1-d4b0-5494-fe38219d16c9
* Version Independent ID: 8b659e63-93b8-a496-bc9b-dce0d9fa8f3c
* Content: [Filter methods for Razor Pages in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/filter?view=aspnetcore-3.1)
* Content Source: [aspnetcore/razor-pages/filter.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/razor-pages/filter.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-razorpages**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #16331.

@Rick-Anderson I've gone for two commits:

1. The DI changes.
2. Some additional tweaks and cleanup.

I usually do the self-review thing, but I don't know how useful this is. Just throw any questions you have at me, or request a self-review and I'll do that to explain the thinking behind suggested changes.

### CODE REVIEW

File: aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs [hunk 1]
```diff
@@ -17,16 +17,16 @@ public SamplePageFilter(IConfiguration config)
...
             ProcessUserAgent.Write(context.ActionDescriptor.DisplayName,
                                    "SamplePageFilter.OnPageHandlerSelected",
-                                    value, key.ToString());
+                                   value, key.ToString());
         }
 
         public void OnPageHandlerExecuting(PageHandlerExecutingContext context)
         {
-            Debug.WriteLine("Global sync PageHandlerExecutingContext called.");
+            Debug.WriteLine("Global sync OnPageHandlerExecuting called.");
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs]: I've been using a variant of
```

            MyDebug.Write(MethodBase.GetCurrentMethod(), context.HttpContext.Request.Path);

public static class MyDebug
    {
        public static void Write(MethodBase m, string path)
        {
            Debug.WriteLine(m.ReflectedType.Name + "." + m.Name + " " +
                path);
        }
    }
```

Not sure if it's worth using here.

File: aspnetcore/razor-pages/filter/3.1sample/PageFilter/Pages/Index.cshtml.cs [hunk 2]
```diff
@@ -17,29 +17,23 @@ public IndexModel(IConfiguration config)
...
 
         public async override Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, 
                                                                PageHandlerExecutionDelegate next)
         {
-            var key = _config.GetValue(typeof(string), "UserAgentID");
+            var key = _config["UserAgentID"];
             context.HttpContext.Request.Headers.TryGetValue("user-agent", out StringValues value);
             ProcessUserAgent.Write(context.ActionDescriptor.DisplayName,
-                                   "/IndexModel-OnPageHandlerSelectionAsync",
+                                   "/IndexModel-OnPageHandlerExecutionAsync",
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/filter/3.1sample/PageFilter/Pages/Index.cshtml.cs]: `ProcessUserAgent.Write` could probably benefit from

```
public static class MyDebug
    {
        public static void Write(MethodBase m, string path)
        {
            Debug.WriteLine(m.ReflectedType.Name + "." + m.Name + " " +
                path);
        }
    }
```

**Comment by Rick-Anderson** [comment]
merged so I could merge live. These improvements too important to wait.

File: aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs (hunk 1 above)
Comment (serpent5) [reply aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs]: Yeah, it's nice to ensure that bad copy/pastes are avoided. Overall, I think I would use `ILogger` for these messages and take advantage of the automatic log category stuff for the context. Combining that with `[CallerMemberName]` would remove the need for reflection too.

Comment (Rick-Anderson) [reply aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs]: `ILogger ` is not available to every filter in this sample.

Comment (serpent5) [reply aspnetcore/razor-pages/filter/3.1sample/PageFilter/Filters/SamplePageFilter.cs]: Makes sense.

**Pull request merged**